onnx/test/shape_inference_test.py

```diff
@@ -383,6 +383,18 @@ def test_expand_dynamic_shape(self) -> None:
             graph,
             [make_tensor_value_info('y', TensorProto.INT32, (None, 2, None))])
 
+    def test_expand_symbolic_shape(self) -> None:
+        graph = self._make_graph(
+            [('x', TensorProto.INT32, (1, 2, None)),
+             ('shape', TensorProto.INT64, ('unk__0',))],
+            [make_node("Expand", ['x', 'shape'], ['y'])],
+            [],
+            initializer=[])
+        # if giving a symbolic shape, Expand should not infer any shape or rank inference
+        self._assert_inferred(
+            graph,
+            [make_tensor_value_info('y', TensorProto.INT32, ())])
```

Comment:
Is that `()` mean an empty shape (rank 0) or no shape?

Reply:
Ideally it should be `None` or something like that

Reply:
Good question. It's tricky here because it seems that both shape (rank 0) and no shape represent in the same way (empty shape)... Let me investigate more and update the test

Reply:
I think we should supply `None` here when creating the value_info. But the comparison should be fixed at https://github.com/onnx/onnx/blob/d824ce38f935b83f724628a254cb855a95180778/onnx/test/shape_inference_test.py#L76 to check if one has no shape the other has no shape also.

Reply:
Yes! I am exactly doing the same thing by adding:
```
assert vi_type.tensor_type.HasField('shape') == inferred_vi_type.tensor_type.HasField('shape')
```
I have tried that rc2 cannot pass this test, but with my fix in this PR, it can pass. Thank you!